redundant-bmc/src/manager.cpp

```diff
@@ -187,8 +186,8 @@ sdbusplus::async::task<std::optional<role_determination::RoleInfo>>
     // An unprovisioned BMC cannot be active.
     if (!services->getProvisioned())
     {
-        lg2::info("Role = passive because BMC not provisioned");
-        co_return RoleInfo{Role::Passive, ErrorCase::notProvisioned};
+        co_return RoleInfo{Role::Passive, ErrorCase::notProvisioned,
+                           "BMC not provisioned"};
```

Comment:
Can we use getNoRedundancyDescription() to keep the strings all in one place?

Reply:
Yea, I can change it to do it the same way as the RedundancyReasons was done.

Reply:
Simplified things to just have the Role and the RoleReason enum, here, and then added function to get the description and if the reason is an error case.